kamelet-support/src/main/java/io/kaoto/backend/KamelPopulator.java

```diff
@@ -504,6 +504,10 @@ private void assignParameters(final Step step, final MarshalFlowStep marshal) {
         marshal.getDataFormat().setProperties(new HashMap<>());
 
         for (Parameter p : step.getParameters()) {
+            if (null == p.getValue()) {
+                continue;
+            }
+
             if ("dataformat".equalsIgnoreCase(p.getId())) {
                 marshal.getDataFormat().setFormat(p.getValue().toString());
```

Comment:
This could be replaced by String.valueOf(p.getValue()) but I think I like your solution more.

Reply:
Argh, I didn't see this :grimacing: , I can change it to incorporate it, the part that I like is that we wouldn't have that extra `if` statement there :ok_hand: